sbin/Makefile.am

```diff
@@ -55,10 +52,65 @@ if ENABLE_VNETBUILD
 scripts += vnetbuild
 endif
 
+
+if FILENAME_VERSIONS
+
+install-exec-hook:
+if ENABLE_FIREHOL
+	cd $(DESTDIR)$(sbindir) && \
+		mv -f firehol firehol-@PACKAGE_VERSION@ && \
+		$(LN_S) firehol-@PACKAGE_VERSION@ firehol
+endif
+if ENABLE_FIREQOS
+	cd $(DESTDIR)$(sbindir) && \
+		mv -f fireqos fireqos-@PACKAGE_VERSION@ && \
+		$(LN_S) fireqos-@PACKAGE_VERSION@ fireqos
+endif
+if ENABLE_LINK_BALANCER
+	cd $(DESTDIR)$(sbindir) && \
+		mv -f link-balancer link-balancer-@PACKAGE_VERSION@ && \
+		$(LN_S) link-balancer-@PACKAGE_VERSION@ link-balancer
+endif
+if ENABLE_UPDATE_IPSETS
+	cd $(DESTDIR)$(sbindir) && \
+		mv -f update-ipsets update-ipsets-@PACKAGE_VERSION@ && \
+		$(LN_S) update-ipsets-@PACKAGE_VERSION@ update-ipsets
+endif
+if ENABLE_VNETBUILD
+	cd $(DESTDIR)$(sbindir) && \
+		mv -f vnetbuild vnetbuild-@PACKAGE_VERSION@ && \
+		$(LN_S) vnetbuild-@PACKAGE_VERSION@ vnetbuild
+endif
```

Comment:
I implemented what is suggested in the automake manual.

I don't mind changing to put the scripts under `/usr/libexec` instead but I don't think your example can work exactly as is. What is `readline` in this context? I don't have any such command on my system or as a bash builtin.

`dirname` is not a builtin, so where does it get configured? The existing PROGRAM_DIR logic is bash-specific but gets around the problem.

Either way, if the program is run using the link I think $0 will refer to the directory with the link, not the directory with the link.

So I think I can achieve most of what you are asking for but I believe then what needs installing to /sbin is not symlinks but tiny placeholder programs e.g.

~~~~~
#!/bin/sh
exec  /usr/libexec/firehol/firehold-@PACKAGE_VERSION@/firehol "$@"
~~~~~

Which is what @ktsaou asked to avoid.